src/components/WeatherComponent.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { StyleSheet, View, ScrollView } from 'react-native';
+import { Warning, WarningType } from '@components/Warning';
+import TimeRelated from '@components/TimeRelated';
+import { ITimeProps } from '@components/TimeRelated';
+import { DateRelated, IDataProps } from '@components/DateRelated';
+import colors from '@styles/colors';
+import { ExtraInfo, IExtraInfoProps } from './ExtraInfo';
+
+export interface IWeatherComponent {
+    warningType: WarningType;
+    arrayOfParameterTimeRelated: Array<ITimeProps>;
+    arrayOfParameterDateRelated: Array<IDataProps>;
+    arrayOfParameterExtraInfo: Array<IExtraInfoProps>;
+}
+
+const SetTimeRelatedComponents = (arrayOfParameterTimeRelated: Array<ITimeProps>) => {
+    const ArrayTimeRelatedComponents = arrayOfParameterTimeRelated.map(element => {
+        return (
+            <TimeRelated
+                time={element.time}
+                weatherType={element.weatherType}
+                temperature={element.temperature}
+                key={element.time}
+            />
+        );
+    });
+    return ArrayTimeRelatedComponents;
+};
```

Comment:
First of all we name functions in camelCase so it should start with a lowercase letter.
This function is not a setter, it doesn't change anything, it returns some value so it is a getter.
There is no need to introduce variable, you can return array.map() directly.
As key it is better to do: 
`array.map((element, index) => return <Timerelated key={index} />)`
Also do it for the rest